### DISCUSSION TITLE
Improve testing

### DISCUSSION THREAD

**Discussion by jorgen** [jorgen/json_struct#53] [General]
@DavidKorczynski added fuzzing, and I just added a github workflow for running all the unit tests with clangs Address Sanitizer and Memory Sanitizer. Does anyone else know of anything else to improve the testing of json_struct?

**Comment by DavidKorczynski** [comment]
An option is to run https://github.com/ossf/scorecard on the project and assess what's missing. I'm quite sure there are some missing SAST tools -- which, although, isn't testing may provide some insights into bugs (or the lack of). E.g. CodeQL? https://github.com/github/codeql-action

**Reply by jorgen** [reply]
Ok, that took a while, but I have now enabled some microsoft SAST and CodeQL and scorecard for master branch and pull requests. I also asked PVS-Studio for a license to check it against json_struct and found a couple of things.
